File: ControlFlow/python/SharedDigit.py
def has_shared_digit(number1: int, number2: int):
    if (99 < number1 < 10) or (99 < number2 < 10): return False;

    while number1 > 0:
        value = number1 % 10;
        value2 = number2
        
        while value2 > 0:
            if (value == value2 % 10): return True;
            value2 = int(value2 / 10)
        number1 = int(number1 / 10)
    return False;


print(has_shared_digit(12,13))
print(has_shared_digit(22,13))
print(has_shared_digit(9,13))


def has_shared_digit_2(number1: int, number2: int):
    if (99 < number1 < 10) or (99 < number2 < 10): return False;
    for number in str(number1):
        if number in str(number2):
            return True;
    return False;

print(has_shared_digit_2(12,13))
print(has_shared_digit_2(22,13))
print(has_shared_digit_2(9,13))

File: ControlFlow/java/SharedDigit.java
package ControlFlow.java;

public class SharedDigit {
    public static void main(String[] args) {
        System.out.println(hasSharedDigit(12, 23));// → should return true since the digit 2 appears in both numbers

        System.out.println(hasSharedDigit(9, 99));// → should return false since 9 is not within the range of 10-99

        System.out.println(hasSharedDigit(15, 55));// → should return true since the digit 5 appears in both numbers

        System.out.println(hasSharedDigit(12, 13));
        
    }
    public static boolean hasSharedDigit(int number1, int number2){
        if (99 < number1 || number1 < 10 || 99 < number2 || number2 < 10){
            return false;
        }
        int value;
        while (number1 > 0){
            value = number1 % 10;
            System.out.println(value);
            int value2 = number2;
            while (value2 > 0){
                if (value == value2 % 10){
                    return true;
                }
                value2 /= 10;
            }
            number1 /= 10;
        }
        return false;
    }
}
